synapse/federation/transport/client.py

```diff
@@ -503,7 +504,7 @@ async def get_public_rooms(
         else:
             path = _create_v1_path("/publicRooms")
 
-            args: Dict[str, Any] = {
+            args: Dict[str, Union[str, Iterable[str]]] = {
```

Comment:
Does it make sense to use `MutableMapping` for `QueryParams` (or have a `MutableQueryParams`)?

Reply:
I think I prefer it keeping the (immutable) Mapping. The non-mutable version expresses that functions taking a `QueryParams` won't mutate those parameters. It's inconvenient here if you conditionally build up a parameter dictionary, but that's rare:  we almost always build a parameter dictionary `args` as a dictionary literal and pass it immediately to an HTTP function (without further mutating `args`).

Reply:
That seems fine! 👍